pkg/controller/operators/olm/overrides/inject/inject.go

```diff
@@ -62,6 +62,81 @@ func mergeEnvVars(containerEnvVars []corev1.EnvVar, newEnvVars []corev1.EnvVar)
 	return merged
 }
 
+// InjectEnvFromIntoDeployment injects the envFrom variables
+// into the container(s) of the given PodSpec.
+//
+// If any Container in PodSpec already defines an envFrom variable
+// as any of the provided envFrom then it will be overwritten.
+func InjectEnvFromIntoDeployment(podSpec *corev1.PodSpec, envFromVars []corev1.EnvFromSource) error {
+	if podSpec == nil {
+		return errors.New("no pod spec provided")
+	}
+
+	for i := range podSpec.Containers {
+		container := &podSpec.Containers[i]
+		container.EnvFrom = mergeEnvFromVars(container.EnvFrom, envFromVars)
+	}
+
+	return nil
+}
+
+func mergeEnvFromVars(containerEnvFromVars []corev1.EnvFromSource, newEnvFromVars []corev1.EnvFromSource) (merged []corev1.EnvFromSource) {
+	merged = containerEnvFromVars
+
+	for _, newEnvFromVar := range newEnvFromVars {
+		found := findEnvFromVar(containerEnvFromVars, newEnvFromVar)
+		if found {
+			continue
+		}
+
+		merged = append(merged, newEnvFromVar)
+	}
+
+	return
+}
+
+func findEnvFromVar(EnvFromVar []corev1.EnvFromSource, newEnvFromVar corev1.EnvFromSource) (found bool) {
+	for i := range EnvFromVar {
+		if compareEnvFromVar(EnvFromVar[i], newEnvFromVar) {
+			found = true
+			break
+		}
+	}
+
+	return
+}
+
+func compareEnvFromVar(envFromVar corev1.EnvFromSource, newEnvFromVar corev1.EnvFromSource) (found bool) {
+	compareprefix := newEnvFromVar.Prefix == envFromVar.Prefix
+	var compareConfigMap, compareSecret bool
+
+	// Compare ConfigMapRef
+	if newEnvFromVar.ConfigMapRef == nil && envFromVar.ConfigMapRef == nil {
+		compareConfigMap = true
+	} else if newEnvFromVar.ConfigMapRef != nil && envFromVar.ConfigMapRef != nil {
+		if newEnvFromVar.ConfigMapRef.Optional == nil && envFromVar.ConfigMapRef.Optional == nil {
+			compareConfigMap = newEnvFromVar.ConfigMapRef.LocalObjectReference == envFromVar.ConfigMapRef.LocalObjectReference
+		} else if newEnvFromVar.ConfigMapRef.Optional != nil && envFromVar.ConfigMapRef.Optional != nil {
+			compareConfigMap = newEnvFromVar.ConfigMapRef.LocalObjectReference == envFromVar.ConfigMapRef.LocalObjectReference && *newEnvFromVar.ConfigMapRef.Optional == *envFromVar.ConfigMapRef.Optional
+		} else {
+			compareConfigMap = false
+		}
+	}
```

Comment:
This block of conditionals looks pretty complicated and is a bit difficult to read. I wonder if something like this might be a bit easier to grok:
```go
if newEnvFromVar.ConfigMapRef == nil && envFromVar.ConfigMapRef == nil {
  compareConfigMap = true
} else if newEnvFromVar.ConfigMapRef != nil && envFromVar.ConfigMapRef != nil {
  compareConfigMap = newEnvFromVar.ConfigMapRef.LocalObjectReference == envFromVar.ConfigMapRef.LocalObjectReference
  
  if newEnvFromVar.ConfigMapRef.Optional != nil && envFromVar.ConfigMapRef.Optional != nil {
    compareConfigMap = compareConfigMap && *newEnvFromVar.ConfigMapRef.Optional == *envFromVar.ConfigMapRef.Optional
  }
}
```

Reply:
I _think_ this does the same thing but reduces the overall number of conditionals and is IMO a bit easier to read. I am also willing to concede that the GH PR review UI is making this harder to read than it normally would be

Reply:
Another thing that could be done, if constant time checks are not necessary, is to exit early from the function on any false condition. Right away, I could see this:
```
if newEnvFromVar.Prefix != envFromVar.Prefix {
    return false
}
```
etc.